addons/sale_purchase/models/sale_order.py

```diff
@@ -23,8 +23,8 @@ def _action_confirm(self):
             order.order_line.sudo()._purchase_service_generation()
         return result
 
-    def action_cancel(self):
-        result = super(SaleOrder, self).action_cancel()
+    def _action_cancel(self):
+        result = super()._action_cancel()
```

Comment:
Please make a specific commit for this `FIX` (so that I can backport it in 13+) :).